modules/@apostrophecms/ui/ui/apos/lib/i18next.js

```diff
@@ -23,6 +23,14 @@ export default {
       debug: i18n.debug,
       interpolation: {
         escapeValue: false
+      },
+      appendNamespaceToMissingKey: true,
+      parseMissingKeyHandler (key) {
+        if (key.startsWith(`${this.defaultNS[0]}:`)) {
```

Comment:
This only checks for the default namespace. We must make sure it is not any of the namespaces that were pushed to the browser (think of admin UI for non-core modules). Which can be done, see these loops from the same file, note the `addResourceBundle` calls all of which take an `ns` parameter, wrap that to make note of these and you'll have something to check against that covers all the cases.

```
    for (const [ ns, phrases ] of Object.entries(i18n.i18n[i18n.locale])) {
      i18next.addResourceBundle(i18n.locale, ns, phrases, true, true);
    }
    if (i18n.locale !== i18n.defaultLocale) {
      for (const [ ns, phrases ] of Object.entries(i18n.i18n[i18n.defaultLocale])) {
        i18next.addResourceBundle(i18n.defaultLocale, ns, phrases, true, true);
      }
    }
    if ((i18n.locale !== 'en') && (i18n.defaultLocale !== 'en')) {
      for (const [ ns, phrases ] of Object.entries(i18n.i18n.en)) {
        i18next.addResourceBundle('en', ns, phrases, true, true);
      }
    }
```

Reply:
The bug only occurs when the key has no namespace and is therefore using the default namespace. If it's an actual l10n key registered on a namespace then it continues through the normal i18n flow and is displayed correctly as the intended text.

If we did apply this to every namespace, then I think we would break localization since it would hit i18next for conversion without the namespace information.

Reply:
OK so this code runs only if the key is not found. I get that this doesn't break other namespaces. I still don't quite get what's going on with this if statement though, can you explain?

Reply:
Also how will this fix play out if the argument *is* a namespaced key, but the translation just hasn't been written yet?

Reply:
Normally i18next displays unrecognized l10n keys by themselves. This includes strings never meant to be keys (where l10n isn't actively in use), e.g., "Not meant to be localized."

With `appendNamespaceToMissingKey: true`, both the key and the namespace are displayed. Example: "translation:Not meant to be localized"). That means that strings containing colons will be passed through fully and not chopped at the colon (such as URLs).

When no namespace is provided, the default namespace is used ("translation," by default). If a namespace is used, i18next will attempt to find the key in the namespace and display the proper value.

We don't want to print something like "translation:Not meant to be localized." So if we see the "key" starts with the default namespace, we slice that off the string and pass the rest of the string along to be displayed.

Reply:
> Also how will this fix play out if the argument is a namespaced key, but the translation just hasn't been written yet?

It will use the fallback locales as normal. Removing the Spanish `managePages` key entry will use the English entry.